addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -398,18 +387,44 @@ function registerBackendAndFrontendTour(name, options, steps) {
  * @param elementName {string} the element to search
  * @param searchNeeded {Boolean} if the widget is a m2o widget and a search is needed
  */
-function selectElementInWeSelectWidget(widgetName, elementName, searchNeeded = false) {
-    const steps = [clickOnElement(`${widgetName} toggler`, `we-select[data-name=${widgetName}] we-toggler`)];
-
+function selectElementInWeSelectWidget(
+    widgetName,
+    elementName,
+    searchNeeded = false
+) {
+    const we_select = `we-select[data-name=${widgetName}]`;
+    const steps = [
+        {
+            content: `Clicking on the ${widgetName} toggler in vue to select ${elementName}`,
+            trigger: `${we_select} we-toggler`,
+            run: "click",
+        },
+    ];
     if (searchNeeded) {
         steps.push({
             content: `Inputing ${elementName} in m2o widget search`,
-            trigger: `we-select[data-name=${widgetName}] div.o_we_m2o_search input`,
-            run: `text ${elementName}`
+            trigger: `${we_select} div.o_we_m2o_search input`,
+            run: `text ${elementName}`,
         });
     }
-    steps.push(clickOnElement(`${elementName} in the ${widgetName} widget`,
-        `we-select[data-name=${widgetName}] we-button:contains(${elementName})`));
+    steps.push({
+        content: `Clicking on the ${elementName} in the ${widgetName} widget`,
+        trigger: `${we_select} we-button:contains("${elementName}")`,
+        run: "click",
+    });
+    steps.push({
+        content:
+            "Check we-select is set and wait a delay before continue the tour",
+        trigger: `${we_select}:contains(${elementName})`,
+        async run() {
+            // fix underterministic error.
+            // When we-select is used twice a row too fast, the second we-select may not open.
+            // The first toggle is open, we click on it and almost at the same time, we click on the second one.
+            // There may be confusion with the active class.
+            // Add a delay before continue the tour solves this problem.
+            await new Promise((resolve) => setTimeout(resolve, 300));
```

Comment:
@qsm-odoo This tour is skipped in all odoo versions because of this. Pierre is currently trying to unskip all skipped tests. I hope you understand that his strategy is restricted to the tour itself, not necessarily fixing the underlying codebase (when the issue comes from there, which isn't always the case) except if that's trivial. However, you're right that a TODO would be nice here, and you (or someone else?) should be aware of that issue and add it on his todo list.

@pipu-odoo could you add a todo above the step?

Reply:
> you (or someone else?) should be aware of that issue and add it on his todo list

Yes. We are spending a ridiculous amount of time fixing tour issues, most of the time by trying not to change the tour. If we have to guess that tour delays were added, that main website utils were moved/changed, that the tour system was changed, ... that's just more time we lose :shrug: 

Anyways like I said, I am only expecting a TODO here and potentially a ping somewhere about it. I don't know about this util in particular in the first place, even less so about the sale tour :shrug: And trust me, it's not like I enjoy being the annoying guy :weary: